### ISSUE TITLE
Ability to suppress "AWSHistory is deprecated and will be removed in the next major release." warning during VSCode debugging?

### ISSUE THREAD

**Issue by jakauppila** [aws/aws-tools-for-powershell#364]
### Describe the bug

When debugging a Powershell script via VSCode, I'm getting the following log entry written to my terminal quite often.

`AWSHistory is deprecated and will be removed in the next major release. Instead, use '-Select *' when it is necessary to return the entire service response as cmdlet output.`

I'm assuming it's due to the variable watching that vscode does and since it is retrieving $AWSHistory, it's logging that warning.

![image](https://github.com/user-attachments/assets/27ebfd1f-b00f-4a65-9417-15d30feffe5b)

Is there a way to suppress that warning? If not, could one be added? 

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Expected Behavior

I would like to be able to suppress the log warning since I'm not explicitly accessing the variable so don't care to see it.

### Current Behavior

When debugging a Powershell script in VSCode that loads any of the AWS.Tools.* and adding a breakpoint to step through it, it will log the warning once an AWS function is called and $AWSHistory is populated

### Reproduction Steps

Within VSCode

```powershell
Install-Module AWS.Tools.Common

Write-Output 'Hello'

Set-DefaultAWSRegion 'us-east-2'

Write-Output 'World'
```

Put a breakpoint at the second Write-Output with F9 and press F5 to run the script.

### AWS Tools for PowerShell version used

AWS.Tools.* 4.1.704

### PowerShell version used

```
Name                           Value
----                           -----
PSVersion                      7.4.4
PSEdition                      Core
GitCommitId                    7.4.4
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Operating System and version

Windows 11

**Comment by ashishdhingra** [comment]
The mentioned warnings are emitted in [AWSHistoryCmdlets.cs](https://github.com/aws/aws-tools-for-powershell/blob/8ae756c11ea510cc8f935cd1329b8e180c27ac76/modules/AWSPowerShell/Common/AWSHistoryCmdlets.cs#L33C42-L33C50) using the [WriteWarning()](https://learn.microsoft.com/en-us/dotnet/api/system.management.automation.cmdlet.writewarning?view=powershellsdk-7.4.0#system-management-automation-cmdlet-writewarning(system-string)) method of [PSCmdLet](https://learn.microsoft.com/en-us/dotnet/api/system.management.automation.pscmdlet?view=powershellsdk-7.4.0) class. Per [$WarningPreference](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_preference_variables?view=powershell-7.4#warningpreference), the value `SilentlyContinue` should not display warning message and the execution should continue.

Tried explicitly setting `$WarningPreference="SilentlyContinue"`, but it doesn't appear to work in Visual Studio Code. This appears to be a general VS Code question rather than AWS PowerShell issue.

**Comment by Rom265** [comment]
This is super annoying. The issue seems to compound. I just started to step through code and before it ran anything, it printed that line 36 times. I did just open a bug report with the VS Code Powershell Extension team. https://github.com/PowerShell/vscode-powershell/issues/5106

**Comment by JustinGrote** [comment]
https://github.com/aws/aws-tools-for-powershell/blob/e2ccfa5ff4f950d95e5838816e219d109cb2d72f/modules/AWSPowerShell/Common/AWSHistoryCmdlets.cs#L38

This line should be in `BeginProcessing` not `ProcessRecord`, it will emit once for every object passed to AWSHistory, which is not what I think you want.

e.g. 1..100 will cause this to emit 100 times.

**Comment by afroz429** [comment]
Thanks for reporting the issue. Currently, there is no way to disable the warning. We will issue a fix soon so that the warning is displayed only once per session instead of displaying the warning each time the `$AWSHistory` variable is evaluated.

**Comment by afroz429** [comment]
This is fixed in `4.1.714`. Please re-open the ticket if you come across any issues. Thanks!
